kubectl-volsync/cmd/migration_create.go

```diff
@@ -36,70 +93,309 @@ var migrationCreateCmd = &cobra.Command{
+	mc.Cluster = xcr.Cluster
+
+	sCapacity, err := cmd.Flags().GetString("capacity")
 	if err != nil {
 		return err
 	}
-	if len(capacity) > 0 {
-		if _, err := resource.ParseQuantity(capacity); err != nil {
+	if len(sCapacity) > 0 {
+		capacity, err := resource.ParseQuantity(sCapacity)
+		if err != nil {
 			return fmt.Errorf("capacity must be a valid resource.Quantity: %w", err)
 		}
+		mc.Capacity = &capacity
 	}
-	// The PVC name must be specified, and it needs to be in the right format
-	pvcname, err := cmd.Flags().GetString("pvcname")
+
+	accessMode, err := cmd.Flags().GetString("accessmodes")
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to fetch access mode, %w", err)
 	}
-	if _, err := ParseXClusterName(pvcname); err != nil {
-		return err
+
+	if v1.PersistentVolumeAccessMode(accessMode) != v1.ReadWriteOnce &&
+		v1.PersistentVolumeAccessMode(accessMode) != v1.ReadOnlyMany &&
+		v1.PersistentVolumeAccessMode(accessMode) != v1.ReadWriteMany &&
+		v1.PersistentVolumeAccessMode(accessMode) != v1.ReadWriteOncePod {
+		return fmt.Errorf("unsupported access mode: %v", accessMode)
+	}
+	accessModes := []v1.PersistentVolumeAccessMode{v1.PersistentVolumeAccessMode(accessMode)}
+	mc.AccessModes = accessModes
+
+	storageClass, err := cmd.Flags().GetString("storageclass")
+	if err != nil {
+		return fmt.Errorf("failed to fetch storageClass, %w", err)
+	}
+	if storageClass == "" {
+		mc.StorageClassName = nil
+	} else {
+		mc.StorageClassName = &storageClass
+	}
+	// For migration use case, only Copymethod="Direct" is supported
+	mc.CopyMethod = volsyncv1alpha1.CopyMethodDirect
+
+	serviceType, err := cmd.Flags().GetString("servicetype")
+	if err != nil {
+		return fmt.Errorf("please provide service type, err = %w", err)
+	}
+
+	if v1.ServiceType(serviceType) != v1.ServiceTypeClusterIP &&
+		v1.ServiceType(serviceType) != v1.ServiceTypeLoadBalancer {
+		return fmt.Errorf("unsupported service type: %v", v1.ServiceType(serviceType))
+	}
+	mc.ServiceType = (*v1.ServiceType)(&serviceType)
+	mc.RDName = mc.Namespace + "-" + mc.PVCName + "-migration-dest"
+
+	return nil
+}
+
+//nolint:funlen
+func (mc *migrationCreate) newMigrationRelationshipDestination(ctx context.Context) error {
+	mrd := mc.mr.data.Destination
+
+	// Assign the values from migrationCreate built after parsing cmd args
+	mrd.RDName = mc.RDName
+	mrd.PVCName = mc.PVCName
+	mrd.Namespace = mc.Namespace
+	mrd.Cluster = mc.Cluster
+
+	if mc.PVC = mc.getDestinationPVC(ctx); mc.PVC == nil {
+		if mc.Capacity == nil {
+			return fmt.Errorf("capacity arg must be provided")
+		}
+		mrd.Destination.Capacity = mc.Capacity
+
+		// Default "AccessMode = ReadWriteOnce" if not provided
+		mrd.Destination.AccessModes = mc.AccessModes
+
+		if mc.StorageClassName == nil {
+			klog.Infof("Storage class is not specified as part of cmd arg, binding to default storage class")
+			mrd.Destination.StorageClassName = nil
+		} else {
+			mrd.Destination.StorageClassName = mc.StorageClassName
+		}
+
```

Comment:
Since the operator isn't ever going to create the PVC (we're only supporting copymethod direct), we shouldn't be setting these fields in the ReplicationDestination--- just DestinationPVC.

Reply:
sure